starport/pkg/cosmosclient/cosmosclient.go

```diff
@@ -210,6 +213,10 @@ type Response struct {
 	*sdktypes.TxResponse
 }
 
+func (r *Response) SetCodec(cdc codec.Codec) {
```

Comment:
```suggestion
```

Not a fan of having a public API like this. Users shouldn't need to set this, cosmosclient should be doing all the necessary initializations. 

We should not add public funcs just for testing, instead we can find alternative/better ways.

Reply:
Gonna find this way.

Reply:
Wdyt about making codec a public field?